lib/client.js

```diff
@@ -590,10 +593,25 @@ Request.prototype._getFormData = function(){
  */
 
 Request.prototype.callback = function(err, res){
+  // console.log(this._retries, this._maxRetries)
+  if (this._maxRetries && this._retries++ < this._maxRetries && shouldRetry(err, res)) {
+    var timeout = 0;
+    var responseTimeout = 0;
+    if (err && err.timeout) {
+      if (err.message.indexOf('Response') == 0) {
```

Comment:
That looks odd.

1. You shouldn't need to read the timeout from the error. Read request's properties set by the `timeout()` function directly
2. *if* you needed to identify error thrown by superagent, add a property to it, don't touch the message